cli/concept.md

```diff
@@ -0,0 +1,157 @@
+ 
+***
+
+# Architecture
+![](https://github.com/UnleashedMind/docs/blob/master/cli/images/AmplifyCliConcept.jpg)
+## Cli core and plugins
+The Amplify cli uses `gluegun`. It is highly modularized.  <br/>
+The cli core provides the pluggable platform, most of the cli functionalities are delivered by plugins. <br/> 
+The cli core searches for plugins in the global `node_modules` directory, and its own `node_modules` directory.  <br/>
+Plugins are recognized by the `amplify-` prefix in the package names. <br/>
+Plugins communicate with the cli core, and with each other, through the project metadata. The cli core provides the read and write access to the project metadata for the plugins. The project metadata is stored in file `amplify/backend/amplify-meta.json` in the user project.
+
+## Plugin types
+There are four types of plugins
+- category
+- provider
+- frontend
+- general purpose
+### category plugin
+Recognized by the `amplify-category-` prefix in the package name, a category plugin wraps up the logic to create and manage one category of backend resources in the cloud. It defines the "shape" of the cloud resources based on user (the developer) input, constructs parameters to CRUD cloud resource, and exports relevant cloud resource information to the project metadata. 
+### provider plugin
+Recognized by the `amplify-provider-` prefix in the package name, a provider plugin abstracts the actual cloud resource provider. It wraps up communication details such as access credentials, api invoke and wait logic, and response data parsing etc. and exposes simple interface methods for the category plugins to CRUD cloud resource. 
+### frontend plugin
+Recognized by the `amplify-frontend-` prefix in the package name, a frontend plugin handles a specific type of frontend projects, such as Javascript, Android or iOS projects. Among other things, it provides these functionalities:
+- formats the cloud resource information and writes it to a file at the right location so it can be recognized and consumed by the frontend project. 
+- builds and serves the frontend application locally with backend hot-wired to the cloud resources. 
+- builds and publishes the application (frontend and backend) to its intended users.
+### general purpose plugin
+Recognized by the `amplify-` prefix, without a plugin type decoration, in the package name, a general purpose plugin does not manage any backend resources in the cloud, but provides certain cli commands and/or certain functionalities for the cli core, and other plugins.
+
+## Official plugins
+- amplify-category-analytics
+- amplify-category-api
+- amplify-category-auth
+- amplify-category-function
+- amplify-category-hosting
+- amplify-category-notifications
+- amplify-category-storage
+- amplify-codegen
+- amplify-frontend-javascript
+- amplify-frontend-android
+- amplify-frontend-ios
+- amplify-provider-awscloudformation
+
+## Third party plugin setup
```

Comment:
I think this needs a little more detail. Is this the plugin that Kaustav made a doc for?

Reply:
added